pkg/clustermanager/retrier_client_test.go

```diff
@@ -0,0 +1,191 @@
+package clustermanager_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/clustermanager/mocks"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+func TestApplyKubeSpecFromBytes(t *testing.T) {
+	tt := NewWithT(t)
+	ctrl := gomock.NewController(t)
+	client := mocks.NewMockKubernetesClient(ctrl)
+	cluster := types.Cluster{}
+
+	client.EXPECT().ApplyKubeSpecFromBytes(context.Background(), &cluster, []byte{}).MinTimes(2).Return(errors.New("this is an  error"))
+	retrierClient := clustermanager.NewRetrierClient(client, retrier.NewWithMaxRetries(2, 0))
+
+	err := retrierClient.ApplyKubeSpecFromBytes(context.Background(), &cluster, []byte{})
+	tt.Expect(err).To(HaveOccurred())
```

Comment:
Could we assert the content of the error pls? (in all the tests)

Reply:
For these particular unit tests though, we don't care about the error though. The content of the error would be unit tested in the part that actually calls the retrier. These unit tests only validate that the retrier is working as expected.